dot/parachain/statement-distribution/state_v2.go

```diff
@@ -0,0 +1,206 @@
+	disableBm := make([]bool, len(group))
+	for idx, v := range group {
+		disableBm[idx] = p.isDisabled(v)
+	}
+
+	bm, err := parachaintypes.NewBitVec(disableBm)
+	if err != nil {
+		logger.Criticalf("cannot create a bitvec: %s", err.Error())
+	}
+
+	return bm, err
+}
+
+type localValidatorStore struct {
+	gridTracker any // TODO: use GridTracker implementation
+	active      *activeValidatorState
+}
+
+type activeValidatorState struct {
+	index          parachaintypes.ValidatorIndex
+	groupIndex     parachaintypes.GroupIndex
+	assignments    []parachaintypes.ParaID
+	clusterTracker any // TODO: use cluster tracker implemenetation
+}
+
+type perSessionState struct {
+	sessionInfo parachaintypes.SessionInfo
+	groups      *groups
+	authLookup  map[parachaintypes.AuthorityDiscoveryID]parachaintypes.ValidatorIndex
+	gridView    any // TODO: use SessionTopologyView from statement-distribution grid
+
+	// when localValidator is nil means it is inactive
+	localValidator     *parachaintypes.ValidatorIndex
+	allowV2Descriptors bool
+}
+
+func newPerSessionState(sessionInfo parachaintypes.SessionInfo,
+	keystore keystore.Keystore,
+	backingThreshold uint32,
+	allowV2Descriptor bool,
+) *perSessionState {
+	authlookup := make(map[parachaintypes.AuthorityDiscoveryID]parachaintypes.ValidatorIndex)
+	for idx, ad := range sessionInfo.DiscoveryKeys {
+		authlookup[ad] = parachaintypes.ValidatorIndex(idx)
+	}
+
+	var localValidator *parachaintypes.ValidatorIndex
+	validatorPk, validatorIdx := parachainutil.SigningKeyAndIndex(sessionInfo.Validators, keystore)
+	if validatorPk != nil {
+		localValidator = &validatorIdx
+	}
+
+	return &perSessionState{
+		sessionInfo:        sessionInfo,
+		groups:             newGroups(sessionInfo.ValidatorGroups, backingThreshold),
+		authLookup:         authlookup,
+		localValidator:     localValidator,
+		allowV2Descriptors: allowV2Descriptor,
+		gridView:           nil,
+	}
+}
+
+// supplyTopology sets the topology for the session and updates the local validator
+// Note: we use the local index rather than the `perSessionState.local_validator` as the
+// former may be `Some` when the latter is `None`, due to the set of nodes in
+// discovery being a superset of the active validators for consensus.
+func (s *perSessionState) supplyTopology(topology *grid.SessionGridTopology, localIdx *parachaintypes.ValidatorIndex) {
+	// TODO #4373: implement once buildSessionTopology is done
+	// gridView := buildSessionTopology(
+	// 	s.sessionInfo.ValidatorGroups,
+	// 	topology,
+	// 	localIdx,
+	// )
+
+	// s.gridView = gridView
+
+	logger.Infof(
+		"Node uses the following topology indices: "+
+			"index_in_gossip_topology: %d, index_in_parachain_auths: %d",
+		localIdx, s.localValidator)
+}
+
+type peerState struct {
+	view            parachaintypes.View
+	protocolVersion parachainnetwork.ValidationVersion
+	implicitView    map[common.Hash]struct{}
```

Comment:
Why doesn't the field name align with the type?

Type is set of Hash and field name says it's the implicit view. 🤔

Reply:
I didn't understand your complain here whether is related to the naming or with the field purpose, but this field is a helper to track other peers implicit view, so we don't need the full backing implicit view, just a set of hashes is enough to check/verify some infos.

Reply:
please rename the field name to some descriptive name that explains what we store in the field.

Implicit view field name is confusing here as it's not actually storing  implicitView.

Can it be something like 'existsInImplicitView'?

Reply:
`existsInImplicitView` is sounds more like a function name than a field name. The field is under the `peerState` struct which holds informations about other peers like supported protocol, latest finalized block and the peer's **implicit view** which makes sense to keep as it is.